src/backend/pagemap.h

```diff
@@ -117,15 +121,32 @@ namespace snmalloc
         bits::ADDRESS_BITS - GRANULARITY_BITS;
       static constexpr size_t ENTRIES = bits::one_at_bit(COVERED_BITS);
 
-      // TODO request additional space, and move to random offset.
+      static constexpr size_t REQUIRED_SIZE = ENTRIES * sizeof(T);
 
-      auto new_body_untyped = Pal::reserve(ENTRIES * sizeof(T));
+#ifdef SNMALLOC_CHECK_CLIENT
+      // Allocate a power of two extra to allow the placement of the
+      // pagemap be difficult to guess.
+      size_t additional_size = bits::next_pow2(REQUIRED_SIZE) * 2;
+      size_t request_size = REQUIRED_SIZE + additional_size;
+#else
+      size_t request_size = REQUIRED_SIZE;
+#endif
 
-      auto new_body = reinterpret_cast<T*>(new_body_untyped);
+      auto new_body_untyped = Pal::reserve(request_size);
```

Comment:
I am not sure if these should not be PAL instead of Pal. If the former then maybe it could be considered to differentiate the naming a bit more (maybe a refactor to remove the usage of Pal in other places in favour of DefaultPal?)

Reply:
Good catch. I will Remove `Pal` in a subsequent PR.